controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -139,9 +139,7 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		"app": "noobaa",
 	}
 
-	if sc.Spec.AllowRemoteStorageConsumers {
-		util.AddAnnotation(nb, "MulticloudObjectGatewayProviderMode", "true")
-	}
+	util.AddAnnotation(nb, "MulticloudObjectGatewayProviderMode", "true")
```

Comment:
does any change to this needs a touch in Noobaa? Let's note such instances directly on corresponding Jira for posterity.

can you get more info on this? is removing this check makes MCG behave differently than internal mode, which we don't want.

Reply:
Adding this annotation sets the "RESTRICT_RESOURCE_DELETION" env variable in noobaa. This ensures that the bucket can be deleted only by the owner of the bucket and not by any other noobaa account.
cc: @ezio-auditore

Reply:
How would that affect users who deletes buckets using S3? or using the new s3 bowser experience?

Reply:
Correction: It is meant to restrict the deletion of a resource (backingstore\namespacestore) and only allow it to be deleted by the resources creator or system admin.